### PR TITLE
Add support for SFTP logging

### PR DESCRIPTION
**Pull request by ezkl**
Adds support for configuring [SFTP log streaming](https://docs.fastly.com/en/guides/log-streaming-sftp) ([API docs](https://developer.fastly.com/reference/api/logging/sftp/)) to provider.

cc: @phamann

### CODE REVIEW

File: fastly/block_fastly_service_v1_logging_sftp_test.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+		comment = "tf-sftp-logging"
+	}
+
+	backend {
+		address = "aws.amazon.com"
+		name		= "amazon docs"
+	}
+
+	logging_sftp {
+		name						= "sftp-endpoint"
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_sftp_test.go:267]: I see weird formatting here. I could probably quickly fix this.

**Review by mccurdyc — APPROVED** [review]
LGTM!

Just some weird formatting in the tests' TF blocks.

File: fastly/block_fastly_service_v1_logging_sftp.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_sftp.go:123]: @mccurdyc TODO: update to use new go-fastly version added in (https://github.com/terraform-providers/terraform-provider-fastly/pull/248) and add MessageType field.

Comment (mccurdyc) [reply fastly/block_fastly_service_v1_logging_sftp.go:123]: done

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

File: website/docs/r/service_v1.html.markdown [hunk 3]
```diff
@@ -85,8 +85,7 @@ resource "aws_s3_bucket" "website" {
 }
 ```
 
-Basic usage with [custom
-VCL](https://docs.fastly.com/vcl/custom-vcl/uploading-custom-vcl/):
+Basic usage with [custom VCL](https://docs.fastly.com/guides/vcl/uploading-custom-vcl):
```
Comment (mccurdyc) [inline website/docs/r/service_v1.html.markdown]: Not sure about this one.

**Comment by mccurdyc** [comment]
Force-pushed squashing commits addressing self-review.

**Review by phamann — APPROVED** [review]
👍 LGTM

**Pull request merged**